### ISSUE TITLE
AppStore delivery validation fails 

### ISSUE THREAD

**Issue by bamx23** [openbakery/gradle-xcodePlugin#298]
The first problem is with a strange libswiftRemoteMirror.dylib in the root of **.app**:

> Invalid Bundle Structure - The binary file 'MyApp.app/libswiftRemoteMirror.dylib' is not permitted. Your app can't contain standalone executables or libraries, other than the CFBundleExecutable of supported bundles. Refer to the Bundle Programming Guide at https://developer.apple.com/go/?id=bundle-structure for information on the iOS app bundle structure.

The second one is here:

> Invalid Swift Support - The files libswiftDarwin.dylib, libswiftDispatch.dylib, libswiftCoreGraphics.dylib, libswiftCoreData.dylib, libswiftUIKit.dylib, libswiftContacts.dylib, libswiftCore.dylib, libswiftFoundation.dylib, libswiftCoreImage.dylib, libswiftObjectiveC.dylib aren't at the expected location /Payload/MyApp.app/Frameworks. Move the file to the expected location, rebuild your app using the current public (GM) version of Xcode, and resubmit it.

I've solved the first one by removing libswiftRemoteMirror.dylib file before packaging.

**Comment by renep** [comment]
Both issues are fixed in the current develop version.
Try:

```
buildscript {
	repositories {
		maven {
			url('http://repository.openbakery.org/')
		}
		mavenCentral()
	}

	dependencies {
			classpath "org.openbakery:xcode-plugin:0.14.2.develop.+"
	}
}
apply plugin: "org.openbakery.xcode-plugin"
```

**Comment by renep** [comment]
I think I will release a 0.14.2 in the next view days ;-)

**Comment by bamx23** [comment]
I'll try it, thank you

**Comment by bamx23** [comment]
And now it hangs on `/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -showBuildSettings`
Here a discussion about this problem with Xcode 8: https://forums.developer.apple.com/thread/50372
Maybe there should be `/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild clean -showBuildSettings` with no difference?

**Comment by sameoldmadness** [comment]
@renep Sorry to bother you, but is there any progress on this one?

**Comment by renep** [comment]
the issue with the `libswiftRemoteMirror.lib` is already in the latest release.
I also updated that a `clean` is included when getting the build settings. You can test this with the latest develop release 0.14.3.develop.+ see:
https://github.com/openbakery/gradle-xcodePlugin#current-develop-version

**Comment by sameoldmadness** [comment]
Awesome, will try it ASAP, thanks.

**Comment by sameoldmadness** [comment]
Worked like a charm, thanks again.

**Comment by renep** [comment]
@bamx23 Can you confirm that this is fixed now?

**Comment by bamx23** [comment]
For 0.14.3.develop.51 now i have this error: 

    Execution failed for task ':package'.
    > Command failed to run (exit code 1): '/usr/bin/codesign --force --sign iPhone Distribution: Me --verbose /path/to/project/build/package/Payload/My.app/Frameworks/libswiftContacts.dylib --keychain /path/to/project/build/codesign/gradle-1476958564402.keychain'
    Tail of output:
    iPhone Distribution: Me: ambiguous (matches "iPhone Distribution: Me" in /Users/me/Library/Keychains/login.keychain-db and "iPhone Distribution: Me" in /path/to/project/build/codesign/gradle-1476958564402.keychain)

I have certificate in login.keychain but I exported it to use with this plugin (certificateURI, certificatePassword). Should I change somethig on my side?

**Comment by renep** [comment]
@bamx23  fixed in the latest version: 0.14.3.354.develop.53
And please open a new issues next time, because we are taking now about three separate issues here.

**Comment by bamx23** [comment]
Sorry for that. It because I can't be sure that the main issue is resolved because of other issues. It's sad that I have to use SHA identities on macOS Siera. But by the way, everything works fine. When do you plan to release 0.14.3?
